### ISSUE TITLE
hparams: why not show all metrics for multi-task?

### ISSUE THREAD

**Issue by geekhch** [tensorflow/tensorboard#5414]
I hope to recoard hparams for multi-task project.
### My Code:
```py
# for the 1st task:
writer.add_hparams(hparams,{"dev_tp_sim": 0.5, "dev_loss_sim": 0.5}, name='hparam')
# for the 2nd task:
writer.add_hparams(hparams,{"dev_tp_nil": 0.5, "dev_loss_nil": 0.5}, name='hparam')
```
### Result with problem (current version: 2.7.0)
![image](https://user-images.githubusercontent.com/32632844/141227276-fedbf11a-42e8-4d58-82b1-180b2a05a065.png)

### Hope to see:
hparams page should show all my metric for every hparam instance, if a instance doesn't have a metric, use NaN fill it.

**Comment by geekhch** [comment]
I guess the reason is that hparam tag aren't able to show NaN metric. I'll try to fix it.

**Comment by UsharaniPagadala** [comment]
@geekhch 
Could you please confirm if this is still an issue.Thanks

**Comment by geekhch** [comment]
> @geekhch Could you please confirm if this is still an issue.Thanks

Hi, sorry I wasn't able to fix it,  it's still an issue.

**Comment by stephanwlee** [comment]
Would you be able to provide more comprehensive repro steps or share a zip file with event files?

**Comment by Kerybas** [comment]
Please have a look there, I believe it is the same issue:
https://github.com/pytorch/pytorch/issues/39250#issuecomment-1304883545
https://github.com/tensorflow/tensorboard/issues/5596
